packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -861,6 +886,50 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
           [queryStateResults, querySubscriptionResults]
         )
       },
+      useUnstable_SuspenseQuery(arg, options) {
+        const querySubscriptionResults = useQuerySubscription(arg, options)
+        const queryStateResults = useQueryState(arg, {
+          selectFromResult: isSkippedQuery(arg, options)
+            ? undefined
+            : noPendingQueryStateSelector,
+          ...options,
+        })
+
+        // We do not suspend if a query is skipped:
+        // @see https://github.com/vercel/swr/pull/357#issuecomment-627089889
+        if (!isSkippedQuery(arg, options)) {
+          if (
+            queryStateResults.isLoading ||
+            (queryStateResults.isFetching && options?.suspendOnRefetch)
+          ) {
+            const pendingPromise = api.util.getRunningOperationPromise<any>(
+              name,
+              arg as unknown as string
+            )
+
+            if (pendingPromise) {
+              throw pendingPromise
```

Comment:
If I'm reading this correctly, this would work like this right now:

* render the component once (loading true, no promise)
* fire the effect to make a request
* render the component again
* throw into suspense

Could we also achieve that without the initial render? The initial render kind of destroys the promise of "data will always be available, you don't need to check for a `isLoading` flag".

Reply:
(that said, I don't want to downplay the work you are putting into this - I'm **very** happy to see you picking this up!)

Reply:
> If I'm reading this correctly, this would work like this right now:
> 
>     * render the component once (loading true, no promise)
> 
>     * fire the effect to make a request
> 
>     * render the component again
> 
>     * throw into suspense
> 
> 
> Could we also achieve that without the initial render? The initial render kind of destroys the promise of "data will always be available, you don't need to check for a `isLoading` flag".

You're right; thank you for you're feedback.

Triggering a prefetch solves this issue:

```diff
+            dispatch(api.util.prefetch(name as any, arg as any, { ifOlderThan: false }))
+
             const pendingPromise = api.util.getRunningOperationPromise<any>(
               name,
               arg as unknown as string
```

Reply:
fixed in https://github.com/reduxjs/redux-toolkit/pull/2149/commits/a6b7f364291f0d1841abcc97716e13dd82712fe8